src/gt4py/backend/templates/dawn_computation.src.in

```diff
@@ -41,17 +41,42 @@ using backend_t = DAWN_GT_BACKEND;
 #include "_dawn_{{ stencil_short_name }}.hpp"
 
 #include <array>
+#include <initializer_list>
 #include <cassert>
 #include <stdexcept>
 
 namespace {{ stencil_unique_name }} {
 
 namespace {
 
+gt::uint_t halos_[] = {
+    GRIDTOOLS_DAWN_HALO_EXTENT, GRIDTOOLS_DAWN_HALO_EXTENT,
+    GRIDTOOLS_DAWN_HALO_EXTENT, GRIDTOOLS_DAWN_HALO_EXTENT, 0, 0
+};
+
+std::array<gt::uint_t, 3> get_min_origin(
+    const std::initializer_list<const std::array<gt::uint_t, 3>>& origins) {
+    std::array<gt::uint_t, 3> min_origin = *origins.begin();
+    for(const auto& origin : origins) {
+        for(int i = 0; i < 3; ++i) {
+            if(origin[i] < min_origin[i])
+                min_origin[i] = origin[i];
+        }
+    }
+    return min_origin;
+}
+
+gridtools::dawn::domain make_domain(const std::array<gt::uint_t, 3> &in_size,
+                                    const std::array<gt::uint_t, 3> &min_origin) {
+    std::array<gt::uint_t, 3> size;
+    for (int i = 0; i < 3; ++i) {
+      if(min_origin[i] < halos_[i*2])
+          halos_[i*2] = halos_[i*2+1] = min_origin[i];
+      size[i] = halos_[i*2] + in_size[i] + halos_[i*2+1];
+    }
 
-gridtools::dawn::domain make_domain(const std::array<gt::uint_t, 3> &size) {
     gridtools::dawn::domain d{size};
-    d.set_halos(0, 0, 0, 0, 0, 0);
+    d.set_halos(halos_[0], halos_[1], halos_[2], halos_[3], halos_[4], halos_[5]);
```

Comment:
Is this working also working if the buffer is shifted when the user sets a different origin?